libgalois/src/MemorySupervisor.cpp

```diff
@@ -16,19 +19,28 @@ getTotalSystemMemory() {
 const std::string sanity_str = "memory manager sanity";
 const std::string oversubscribed_str = "memory manager oversubscribed";
 const std::string unregister_str = "memory manager unregister";
-}  // anonymous namespace
 
 void
-katana::MemorySupervisor::LogState(const std::string& str) {
+LogState(const std::string& str, count_t active, count_t standby) {
   katana::GetTracer().GetActiveSpan().Log(
       str, {
-               {"active", active_},
-               {"standby", standby_},
+               {"active", active},
+               {"standby", standby},
            });
 }
 
 void
-katana::MemorySupervisor::Sanity() {
+KillCheck(katana::MemoryPolicy* policy, count_t active, count_t standby) {
+  if (policy->KillSelfForLackOfMemory(active, standby)) {
+    LogState(oversubscribed_str, active, standby);
+    KATANA_LOG_FATAL("out of memory");
```

Comment:
Should we be broadcasting an error here? I thought one of the advantages of killing the process ourselves was that we could do cleanup and not hang when one host got oversubscribed.

Reply:
Yes, this is messy.  Is there an example of the clean, distributed shutdown you are talking about?

Reply:
@roshandathathri said it was ok.  Ya neat freak.